### ISSUE TITLE
Overall timeout in Gmail tests

### ISSUE THREAD

**Issue by rrrooommmaaa** [FlowCrypt/flowcrypt-browser#5095]
Hi @sosnovsky 

As gmail live tests are run one by one, we can easily exceed the limit of 30 minutes (the limit is certainly exceeded on my local machine).

I tried to increase the timeouts with this commit https://github.com/FlowCrypt/flowcrypt-browser/pull/5069/commits/269b55fadfef2035f8663bcd608e49eb5a1a0fbf on my working branch, but still Semaphore reports a failure after 30 minutes.
How can we extend it?

We can also move some of the tests to mock environment, but we need to figure out what to move and what to leave (we should keep track of the actual Gmail message container classes etc.)

**Comment by sosnovsky** [comment]
> I tried to increase the timeouts with this commit https://github.com/FlowCrypt/flowcrypt-browser/commit/269b55fadfef2035f8663bcd608e49eb5a1a0fbf on my working branch, but still Semaphore reports a failure after 30 minutes.
> How can we extend it?

You can set bigger timeout in semaphore config - https://github.com/FlowCrypt/flowcrypt-browser/blob/master/.semaphore/semaphore.yml#L63

But I think it should be possible to improve live tests speed, as earlier they finished even in 20 minutes. Do you know what caused current time increase?

Also currently `multiple compose windows, saving/opening compose draft` tests fails almost on each run and it's execution takes too much time. Probably fixing it will allow all tests to finish in less than 30 minutes

**Comment by rrrooommmaaa** [comment]
> But I think it should be possible to improve live tests speed, as earlier they finished even in 20 minutes. Do you know what caused current time increase?

No, I don't know this. So far, if the test suite times out, there isn't a way to see which tests passed, which failed etc., is there?

**Comment by rrrooommmaaa** [comment]
> For example here - https://flowcrypt.semaphoreci.com/workflows/e42fb374-85c9-44df-8f16-9db309f4421a?pipeline_id=a474f8e7-a3e9-42ff-abd7-10a4737ad8b4 all live tests passed in 22 minutes, so 30 minutes timeout should enough when all live tests work well

Most of the time is spent on the log-in sequence, so it's not an accurate way to benchmark GmailElementReplacer or Compose windows, etc.
For this we can move some tests to the mock environment

**Comment by sosnovsky** [comment]
> No, I don't know this. So far, if the test suite times out, there isn't a way to see which tests passed, which failed etc., is there?

It's possible to see passed tests after disabling Semaphore test reports (we can disable them for live tests - https://github.com/FlowCrypt/flowcrypt-browser/blob/master/package.json#L82, as they're executed 1-by-1 and we can find which test caused timeout)

> Most of the time is spent on the log-in sequence, so it's not an accurate way to benchmark GmailElementReplacer or Compose windows, etc.
> For this we can move some tests to the mock environment

I think it's ok to have current suite of live tests, as it tests base functionality of FlowCrypt (which should work well on live gmail) and takes around 20 minutes when everything works well. Just need to find how we can make them more stable.
Or maybe you noticed some tests which don't need live gmail?

**Comment by rrrooommmaaa** [comment]
> Also currently `multiple compose windows, saving/opening compose draft` tests fails almost on each run and it's execution takes too much time.

Yes, a bit complicated case with drafts, I'll add additional checks and more descriptive output in PR #5088

**Comment by rrrooommmaaa** [comment]
> For example here - https://flowcrypt.semaphoreci.com/workflows/e42fb374-85c9-44df-8f16-9db309f4421a?pipeline_id=a474f8e7-a3e9-42ff-abd7-10a4737ad8b4 all live tests passed in 22 minutes, so 30 minutes timeout should enough when all live tests work well

They also succeeded this morning without relevant updates, and then failed again on re-run, so this is just a matter of interference. 
![image](https://user-images.githubusercontent.com/23335944/234661699-394961d0-b46a-4e2e-9da4-d52ecf40ffab.png)

**Comment by rrrooommmaaa** [comment]
We should run draft-related live tests (the ones that read and check drafts) one by one...
Decryption-only tests etc. can be run in parallel and it turns out they actually do already.
If all are run one by one, then the run time will certainly exceed 30 minutes

**Comment by rrrooommmaaa** [comment]
Also online-offline tests should not be run individually
![image](https://user-images.githubusercontent.com/23335944/234665516-dde18d78-c6c9-4fa7-8658-f0038a2875fe.png)

**Comment by rrrooommmaaa** [comment]
Why are gmail live tests now running smoothly on Semaphore? We didn't merge any related updates, did we?

**Comment by rrrooommmaaa** [comment]
My recent runs on PR #5069 didn't cause any draft-related failures

**Comment by sosnovsky** [comment]
I think there were any related updates, just some changes to mock test API.
Maybe puppeteer update improved tests stability

**Comment by rrrooommmaaa** [comment]
Timeout was increased by PR #5172
So we can close this issue.
